x-pack/legacy/plugins/apm/public/components/shared/LocalUIFilters/index.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import {
+  EuiTitle,
+  EuiSpacer,
+  EuiHorizontalRule,
+  EuiButtonEmpty
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import styled from 'styled-components';
+import { Filter } from './Filter';
+import { useLocalUIFilters } from '../../../hooks/useLocalUIFilters';
+import { TransactionTypeFilter } from './TransactionTypeFilter';
+
+type Props = Parameters<typeof useLocalUIFilters>[0] & {
+  showCount?: boolean;
+} & (
+    | {
+        showTransactionTypeFilter: true;
+        allowEmptyTransactionType?: boolean;
+        transactionTypes: string[];
+      }
+    | {});
+
+const ButtonWrapper = styled.div`
+   {
+    display: inline-block;
+  }
+`;
+
+const LocalUIFilters: React.FC<Props> = props => {
+  const { projection, params, filterNames, showCount = true } = props;
```

Comment:
Just a general observation:
 - why not destructure `props` in the function signature? 
 - I still don't get the appeal of `React.FC`. It's more clunky and it implicitly types `children`. So `<LocalUIFilters>Hey</LocalUIFilters>` is allowed even though it probably shouldn't be

```js
export const LocalUIFilters({ projection, params, filterNames, showCount = true } : Props) => {
...
}
```

Reply:
Because of:

```js
const {
    showTransactionTypeFilter = false,
    allowEmptyTransactionType = false,
    transactionTypes = []
  } = 'showTransactionTypeFilter' in props ? props : {};
```

one could argue that's a bad pattern though, but I want to make sure people don't pass any props that fall into a black hole if showTransactionType filter is going to be false.

However, we're going to remove this anyway (I'll just pass TransactionTypeFilter as a child).

Reply:
Thanks for pointing that out about the React.FC type, I didn't know about its drawbacks (for those reading along, see https://github.com/typescript-cheatsheets/react-typescript-cheatsheet/issues/87#issuecomment-463578132)